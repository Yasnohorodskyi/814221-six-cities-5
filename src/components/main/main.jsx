import React from "react";
import PropTypes from "prop-types";
import CitiesList from "../cities-list/cities-list";
import {Link} from "react-router-dom";


const Main = ({offersNumber}) => {
  return (
    <React.Fragment>
      <div className="page page--gray page--main">
        <header className="header">
          <div className="container">
            <div className="header__wrapper">
              <div className="header__left">
                <a className="header__logo-link header__logo-link--active">
                  <img
                    className="header__logo"
                    src="img/logo.svg"
                    alt="6 cities logo"
                    width="81"
                    height="41"
                  />
                </a>
              </div>
              <nav className="header__nav">
                <ul className="header__nav-list">
                  <li className="header__nav-item user">
                    <Link to="/favorites"
                      className="header__nav-link header__nav-link--profile"
                    >
                      <div className="header__avatar-wrapper user__avatar-wrapper"></div>
                      <span className="header__user-name user__name">
                        Oliver.conner@gmail.com
                      </span>
                    </Link>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </header>

        <main className="page__main page__main--index">
          <h1 className="visually-hidden">Cities</h1>
          <CitiesList offersNumber = {offersNumber}></CitiesList>
        </main>
      </div>
    </React.Fragment>
  );
};

Main.propTypes = {
  offersNumber: PropTypes.number,
  offers: PropTypes.array,
};
export default Main;
