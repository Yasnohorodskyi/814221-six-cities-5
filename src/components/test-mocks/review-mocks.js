export const review = {
  comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  date: `2019-05-08T14:13:56.569Z`,
  id: 1,
  rating: 4,
  user: {
    photo: `/img/avatar-angelina.jpg`,
    id: 4,
    isPro: false,
    name: `Lena`,
  },
};

export const reviews = [
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:53.569Z`,
    id: 1,
    rating: 4,
    user: {
      photo: `/img/avatar-angelina.jpg`,
      id: 4,
      isPro: false,
      name: `Lena`,
    },
  },

  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`,
    id: 2,
    rating: 5,
    user: {
      photo: `/img/avatar-angelina.jpg`,
      id: 34,
      isPro: true,
      name: `Masha`,
    },
  },
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:58.569Z`,
    id: 3,
    rating: 3,
    user: {
      photo: `/img/avatar-angelina.jpg`,
      id: 39,
      isPro: false,
      name: `Masha`,
    },
  },
];

export const reviewsBeforeAdapt = [
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`,
    id: 2,
    rating: 5,
    user: {
      avatar_url: `/img/avatar-angelina.jpg`,
      id: 34,
      is_pro: true,
      name: `Masha`,
    },
  },
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:58.569Z`,
    id: 3,
    rating: 3,
    user: {
      avatar_url: `/img/avatar-angelina.jpg`,
      id: 39,
      is_pro: false,
      name: `Masha`,
    },
  },
];
