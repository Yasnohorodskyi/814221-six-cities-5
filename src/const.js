export const numberOfFavouritesOffers = 2;
export const AuthorizationCodes = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const SortType = {
  POPULAR: `Popular`,
  PRICE_LOW_TO_HIGH: `Price: low to high`,
  PRICE_HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED_FIRST: `Top rated first`,
};
