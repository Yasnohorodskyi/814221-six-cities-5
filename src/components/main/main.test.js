import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";

import {offersCity} from "../test-mocks/offers-mocks";
import {AuthorizationCodes} from "../../const";
import {Main} from "../main/main";

const renderer = new ShallowRenderer();

describe(`<Main /> render`, () => {
  it(`renders correctly with statusAthoridation = AUTH`, () => {
    renderer.render(
        <Main
          authorizationStatus={AuthorizationCodes.AUTH}
          offersCity={offersCity}
        />
    );

    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  it(`renders correctly with statusAthoridation = NO_AUTH`, () => {
    renderer.render(
        <Main
          authorizationStatus={AuthorizationCodes.NO_AUTH}
          offersCity={offersCity}
        />
    );
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
