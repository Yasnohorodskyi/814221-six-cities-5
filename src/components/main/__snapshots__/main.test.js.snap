// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> render renders correctly with statusAthoridation = AUTH 1`] = `
<React.Fragment>
  <div
    className="page page--gray page--main"
  >
    <header
      className="header"
    >
      <div
        className="container"
      >
        <div
          className="header__wrapper"
        >
          <div
            className="header__left"
          >
            <a
              className="header__logo-link header__logo-link--active"
            >
              <img
                alt="6 cities logo"
                className="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            className="header__nav"
          >
            <ul
              className="header__nav-list"
            >
              <li
                className="header__nav-item user"
              >
                <Link
                  className="header__nav-link header__nav-link--profile"
                  to="/favorites"
                >
                  <div
                    className="header__avatar-wrapper user__avatar-wrapper"
                  />
                  <span
                    className="header__user-name user__name"
                  >
                    Oliver.conner@gmail.com
                  </span>
                </Link>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      className="page page--gray page--main page__main--index-empty"
    >
      <h1
        className="visually-hidden"
      >
        Cities
      </h1>
      <Memo(Connect(CitiesList)) />
    </main>
  </div>
</React.Fragment>
`;

exports[`<Main /> render renders correctly with statusAthoridation = NO_AUTH 1`] = `
<React.Fragment>
  <div
    className="page page--gray page--main"
  >
    <header
      className="header"
    >
      <div
        className="container"
      >
        <div
          className="header__wrapper"
        >
          <div
            className="header__left"
          >
            <a
              className="header__logo-link header__logo-link--active"
            >
              <img
                alt="6 cities logo"
                className="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            className="header__nav"
          >
            <ul
              className="header__nav-list"
            >
              <li
                className="header__nav-item user"
              >
                <Link
                  className="header__nav-link header__nav-link--profile"
                  href="#"
                  to="/login"
                >
                  <div
                    className="header__avatar-wrapper user__avatar-wrapper"
                  />
                  <span
                    className="header__login"
                  >
                    Sign in
                  </span>
                </Link>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      className="page page--gray page--main page__main--index-empty"
    >
      <h1
        className="visually-hidden"
      >
        Cities
      </h1>
      <Memo(Connect(CitiesList)) />
    </main>
  </div>
</React.Fragment>
`;
